error during build python bindings for client

I have compiled and installed the he-transformer and its python client, and I find two error during compile. the first is `undefined reference to symbol 'pthread_join@@GLIBC_2.2.5'`, error show as below:
```
1 warning generated.
[ 80%] Building CXX object test/CMakeFiles/unit-test.dir/main.cpp.o
[ 81%] Building CXX object test/CMakeFiles/unit-test.dir/test_he_op_annotations.cpp.o
[ 82%] Building CXX object test/CMakeFiles/unit-test.dir/test_he_plaintext.cpp.o
[ 82%] Building CXX object test/CMakeFiles/unit-test.dir/test_he_tensor.cpp.o
[ 83%] Building CXX object test/CMakeFiles/unit-test.dir/test_he_type.cpp.o
[ 83%] Building CXX object test/CMakeFiles/unit-test.dir/test_he_util.cpp.o
[ 84%] Building CXX object test/CMakeFiles/unit-test.dir/test_node_wrapper.cpp.o
[ 85%] Building CXX object test/CMakeFiles/unit-test.dir/test_he_fusion.cpp.o
[ 85%] Building CXX object test/CMakeFiles/unit-test.dir/test_he_supported_ops.cpp.o
[ 86%] Building CXX object test/CMakeFiles/unit-test.dir/test_propagate_he_annotations.cpp.o
[ 87%] Building CXX object test/CMakeFiles/unit-test.dir/test_encryption_parameters.cpp.o
[ 87%] Building CXX object test/CMakeFiles/unit-test.dir/test_he_seal_executable.cpp.o
[ 88%] Building CXX object test/CMakeFiles/unit-test.dir/test_bounded_relu.cpp.o
[ 88%] Building CXX object test/CMakeFiles/unit-test.dir/test_perf_micro.cpp.o
[ 89%] Building CXX object test/CMakeFiles/unit-test.dir/test_seal.cpp.o
[ 90%] Building CXX object test/CMakeFiles/unit-test.dir/test_protobuf.cpp.o
[ 90%] Building CXX object test/CMakeFiles/unit-test.dir/test_seal_plaintext_wrapper.cpp.o
[ 91%] Building CXX object test/CMakeFiles/unit-test.dir/test_seal_util.cpp.o
[ 92%] Building CXX object test/CMakeFiles/unit-test.dir/test_tcp_message.cpp.o
[ 92%] Building CXX object test/CMakeFiles/unit-test.dir/test_tcp_client.cpp.o
[ 93%] Building CXX object test/CMakeFiles/unit-test.dir/test_ngraph_he_log.cpp.o
[ 94%] Linking CXX executable unit-test
/usr/bin/ld: CMakeFiles/unit-test.dir/test_server_client_HE_SEAL.cpp.o: undefined reference to symbol 'pthread_join@@GLIBC_2.2.5'
/usr/bin/ld: /lib/x86_64-linux-gnu/libpthread.so.0: error adding symbols: DSO missing from command line
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [test/CMakeFiles/unit-test.dir/build.make:841: test/unit-test] Error 1
make[1]: *** [CMakeFiles/Makefile2:591: test/CMakeFiles/unit-test.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```
it seems that  `/usr/bin/ld`  want to link all object into a binary file but don't find the implement of `pthread_join@@GLIBC_2.2.5`. so I suspect that cmake don't pass `-lpthread` to clang during compile. so I add line `SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")` to `he-transformer-master/test/CMakeLists.txt` , then compile passed.
<br/>
<br/>
another error is `file not recognized: file format not recognized`, detailed error messages is:   
```
End of search list.
creating build/lib.linux-x86_64-3.7
/usr/bin/clang++-6.0 -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-x86_64-3.7/root/he-transformer-master/python/pyhe_client/he_seal_client.o build/temp.linux-x86_64-3.7/root/he-transformer-master/python/pyhe_client/pyhe_client.o -L/root/he-transformer-master/build/python/_install//root/he-transformer-master/build/external/lib -lhe_seal_backend -o build/lib.linux-x86_64-3.7/pyhe_client.cpython-37m-x86_64-linux-gnu.so -Wl,-rpath,$ORIGIN/../.. -z noexecstack -z relro -z now
build/temp.linux-x86_64-3.7/root/he-transformer-master/python/pyhe_client/he_seal_client.o: file not recognized: file format not recognized
clang: error: linker command failed with exit code 1 (use -v to see invocation)
error: command '/usr/bin/clang++-6.0' failed with exit status 1
make[3]: *** [python/CMakeFiles/python_client.dir/build.make:62: python/dist] Error 1
make[2]: *** [CMakeFiles/Makefile2:697: python/CMakeFiles/python_client.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:704: python/CMakeFiles/python_client.dir/rule] Error 2
make: *** [Makefile:364: python_client] Error 2
```
after I google this error, it seems a bug of clang compiler, current llvm version is 6, the same error also occurs with llvm 8. and I have a try to g++(current version is `8.3.0`), and compile passed. what I do is change line `export CXX_COMPILER=@CMAKE_CXX_COMPILER@` to `export CXX_COMPILER=/usr/bin/g++` in `he-transformer-master/python/build_wheel.sh.in`.      
the way I solve compile errors seems not the best, so can you tell me how to avoid this two error?

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -115,7 +115,7 @@ target_include_directories(unit-test
                            PRIVATE ${NGRAPH_TF_BUILD_DIR}/ngraph/test/)
 
 target_link_libraries(unit-test PRIVATE ngraph_test_util)
-target_link_libraries(unit-test PRIVATE ngraph libgtest)
+target_link_libraries(unit-test PRIVATE ngraph libgtest pthread)
 target_link_libraries(unit-test PRIVATE he_seal_backend libseal)
 target_link_libraries(unit-test PRIVATE protobuf::libprotobuf)
 

